src/redux/actions/types.tsx

```diff
@@ -160,138 +161,162 @@ export type SchemaActions =
-    };
-
-    export type FetchProjectByIdSuccess = {
-      type: typeof FETCH_PROJECT_BYID_SUCCESS;
-       payload: FetchProjectByIdSuccessPayload;
-        };
-    
-    export type FetchProjectFailure = {
-      type: typeof FETCH_ALLPROJECTS_FAILURE;
-      payload: FetchProjectFailurePayload;
-    };
-
-    export type FetchProjectByIdFailure = {
-      type: typeof FETCH_PROJECT_BYID_FAILURE;
-       payload: FetchProjectByIdFailurePayload;
-       };
-    
-    export type ProjectActions =
-      | FetchProjectRequest
-      | FetchProjectSuccess
-      | FetchProjectFailure;
-
-      
-
-      export interface IGROUPDATA {
-        projectID: string,
-       name:string,
-      desc:string,
-      attributes:string,
-       
-       
-     
-      }
-      
-      export interface GroupState {
-        pending: boolean;
-        groups: IGROUPDATA[];
-        error: string | null;
-      }
-     
-      export interface FetchGroupSuccessPayload {
-        groups: IGROUPDATA[];
-      }
-      
-      export interface FetchGroupFailurePayload {
-        error: string;
-      }
-      
-      export interface FetchGroupRequest {
-        type: typeof FETCH_ALLGROUP_DATA_REQUEST;
-      }
-      
-      export type FetchGroupSuccess = {
-        
-        type: typeof FETCH_ALLGROUP_DATA_SUCCESS;
-        payload: FetchGroupSuccessPayload;
-      };
-    
-      export type FetchGroupFailure = {
-        type: typeof  FETCH_ALLGROUP_DATA_FAILURE;
-        payload: FetchGroupFailurePayload;
-      };
-      
-      export type GroupActions =
-        | FetchGroupRequest
-        | FetchGroupSuccess
-        | FetchGroupFailure;
-
-      export type ProjectByIdActions =
-                | FetchProjectByIdRequest
-                | FetchProjectByIdSuccess
-                | FetchProjectByIdFailure;
\ No newline at end of file
+  groups: number,
+  users: number,
+  initialLoad: number,
+  sync: number,
+}
+export interface IPROJECT {
+  totalSchemas: number,
+  projects: [],
+  projectDetails: IDETAILS;
+  projectId: number;
+  projectName: string;
+  description: string;
+  uid: number;
+}
+
+export interface IPROJECTBYID {
+  uid: number,
+  name: string,
+  description: string
+}
+
+export interface ProjectState {
+  pending: boolean;
+  projects: IPROJECT[];
+  error: string | null;
+}
+
+export interface ProjectByIdState {
+  pending: boolean;
+  projectById: IPROJECTBYID;
+  error: string | null;
+}
+
+export interface FetchProjectSuccessPayload {
+  projects: IPROJECT[];
+}
+
+export interface FetchProjectByIdSuccessPayload {
+  projectById: IPROJECTBYID;
+}
+
+export interface FetchProjectFailurePayload {
+  error: string;
+}
+
+export interface FetchProjectByIdFailurePayload {
+  error: string;
+}
+
+export interface FetchProjectRequest {
+  type: typeof FETCH_ALLPROJECTS_REQUEST;
+}
+
+export interface FetchProjectByIdRequest {
+  type: typeof FETCH_PROJECT_BYID_REQUEST;
+  params: number;
+}
+
+export type FetchProjectSuccess = {
```

Comment:
Before committing code make sure eslint issues are solved and formatting is done